vm/boostenv/main/boostvm.cc

```diff
@@ -50,7 +50,7 @@ BoostVM::BoostVM(BoostEnvironment& environment,
   uuidGenerator(),
   portClosed(false),
   _asyncIONodeCount(0),
-  preemptionTimer(environment.io_service),
+  preemptionTimer(nullptr),
   alarmTimer(environment.io_service),
```

Comment:
What about this timer? Is it only accessed through this VM's thread?

Reply:
Yes. It would be simpler to do the same for preemptionTimer, but then it would not be able to restart itself, and the restart should be done in the vm when the preemption flag is cleared. And it looks difficult because the vm is a real vm, while the other is a boostvm...

Reply:
> And it looks difficult because the vm is a real vm, while the other is a boostvm...

I think that just need adding one method in the interface between the two, but I don't remember the details. It would change semantics from "preempt every millisecond" to "preempt every millisecond+the time for VM to notice" though.

Reply:
Exactly. And this works, so why bother ;-)